src/BuildScriptGeneratorCli/Commands/ExecCommand.cs

```diff
@@ -0,0 +1,113 @@
+        public const string CommandMissingErrorMessage = "A command is required.";
+        public const string NoToolsDetectedErrorMessage = "No usable tools detected for source directory.";
+
+        [Argument(0, Description = "The source directory.")]
+        public string SourceDir { get; set; }
+
+        [Argument(1, Description = "The command to execute in an app-specific environment.")]
+        public string Command { get; set; }
+
+        internal override int Execute(IServiceProvider serviceProvider, IConsole console)
+        {
+            var logger = serviceProvider.GetRequiredService<ILogger<ExecCommand>>();
+            var env = serviceProvider.GetRequiredService<IEnvironment>();
+            var generator = serviceProvider.GetRequiredService<IBuildScriptGenerator>();
+
+            var shellPath = env.GetEnvironmentVariable("BASH") ?? DefaultBashPath;
+            logger.LogInformation("Using shell {shell}", shellPath);
+
+            var ctx = BuildScriptGenerator.CreateContext(serviceProvider, operationId: null);
+            ctx.DisableMultiPlatformBuild = false;
+            var tools = generator.GetRequiredToolVersions(ctx);
+
+            if (tools.Count == 0)
+            {
+                console.WriteErrorLine(NoToolsDetectedErrorMessage);
+                return ProcessConstants.ExitFailure;
+            }
+
+            int exitCode;
+            using (var timedEvent = logger.LogTimedEvent("ExecCommand"))
+            {
+                var cmd = $"benv {StringExtensions.JoinKeyValuePairs(tools)} {Command}";
+
+                logger.LogInformation("Running {shell} -c {cmd}", shellPath, cmd);
+                if (DebugMode)
+                {
+                    console.WriteLine($"> {shellPath} -c '{cmd}'");
+                }
+
+                exitCode = serviceProvider.GetRequiredService<IScriptExecutor>().ExecuteScript(
+                    shellPath,
+                    new[] { "-c", cmd },
+                    SourceDir,
+                    (sender, args) => console.WriteLine(args.Data),
+                    (sender, args) => console.Error.WriteLine(args.Data));
+
+                timedEvent.AddProperty("exitCode", exitCode.ToString());
+            }
+
+            return exitCode;
+        }
+
+        internal override bool IsValidInput(IServiceProvider serviceProvider, IConsole console)
+        {
+            var logger = serviceProvider.GetRequiredService<ILogger<ExecCommand>>();
+            var options = serviceProvider.GetRequiredService<IOptions<BuildScriptGeneratorOptions>>().Value;
+
+            if (!Directory.Exists(options.SourceDir))
+            {
+                logger.LogError("Could not find the source directory {srcDir}", SourceDir);
+                console.WriteErrorLine(string.Format(SrcDirDoesNotExistErrorMessageFmt, SourceDir));
+                return false;
+            }
+
+            if (string.IsNullOrWhiteSpace(Command))
```

Comment:
I wouldn't error out in this case, just do a no-op. When writing a script that calls this command, it's easier to just no-op when there is no command so the script doesn't fail. I don't think `benv` for instance fails with an empty command